init/pipeline.py

```diff
@@ -1,50 +1,75 @@
 import dlt
-from dlt.sources.helpers import requests
 
+from dlt.sources.helpers.rest_client import paginate
+from dlt.sources.helpers.rest_client.auth import BearerTokenAuth
+from dlt.sources.helpers.rest_client.paginators import HeaderLinkPaginator
 
-@dlt.source
-def source(api_secret_key=dlt.secrets.value):
-    return resource(api_secret_key)
+# This is a generic pipeline example and demonstrates
+# how to use the dlt REST client for extracting data from APIs.
+# It showcases the use of authentication via bearer tokens and pagination.
 
 
-def _create_auth_headers(api_secret_key):
-    """Constructs Bearer type authorization header which is the most common authorization method"""
-    headers = {"Authorization": f"Bearer {api_secret_key}"}
-    return headers
+@dlt.source
+def source(api_secret_key: str = dlt.secrets.value):
+    # print(f"api_secret_key={api_secret_key}")
+    return resource(api_secret_key)
 
 
 @dlt.resource(write_disposition="append")
-def resource(api_secret_key=dlt.secrets.value):
-    headers = _create_auth_headers(api_secret_key)
-
-    # check if authentication headers look fine
-    print(headers)
-
-    # make an api call here
-    # response = requests.get(url, headers=headers, params=params)
-    # response.raise_for_status()
-    # yield response.json()
+def resource(
+    api_secret_key: str = dlt.secrets.value,
+    org: str = "dlt-hub",
+    repository: str = "dlt",
+):
+    yield [
```

Comment:
If we actually yielding data, as an in the previous file version we should keep the comment that explains what's going on. 
e.g "test data for loading validation, delete it once you yield actual data"

Reply:
That's a good point I added it.